### PR TITLE
Tema 1

### PR DESCRIPTION
**Pull request by mindit-coteanu-vlad**
https://drive.google.com/drive/folders/16n2yqZ-O-i5j1X_MHPXq91i9EXl0u2ER

### CODE REVIEW

**Comment by AndreiOprisan94** [comment]
Please remove the **xls** file from the repo, now that it works with **xlsx**

File: CoteanuVlad/src/org/bootcamp/dao/VehicleExcelFileDao.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            }
+
+        } catch (Exception e) {
+            String errorMessage = "Can not create instance of class " + VehicleExcelFileDao.class.getSimpleName();
+            throw new IllegalStateException(errorMessage);
+        }
+    }
+
+    @Override
+    public List<VehicleInfo> getAllVehicles() {
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleExcelFileDao.java]: Hint: Please check the **values** method of the Map class. And look at the constructor documentation for ArratList<>. [Here](https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#ArrayList--)

**Comment by mindit-coteanu-vlad** [comment]
Done 😊

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

File: CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import org.bootcamp.model.VehicleInfo;
+import org.bootcamp.vehicle.Vehicle;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class VehicleJsonFileDao implements VehicleInfoDao {
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java]: Same comment as for the ExcelFileDao. Use **final**, avoid wildcard imports. Make the map attribute final and initialize it at the declaration. Unused **builder** field

File: CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    private final VehicleInfo.Builder vehicleInfoBuilder = VehicleInfo.builder();
+    private Map<String, VehicleInfo> vehicleInfoMap;
+
+    public VehicleJsonFileDao(String filePath) {
+
+        final ObjectMapper objectMapper = new ObjectMapper();
+        List<VehicleInfo> vehicleInfoList;
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java]: Please make final

File: CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        vehicleInfoMap = new HashMap<>();
+        try {
+            final InputStream inputStream = new FileInputStream(new File(filePath));
+            vehicleInfoList = objectMapper.readValue(inputStream, objectMapper.getTypeFactory().constructCollectionType(List.class, VehicleInfo.class));
+            for (VehicleInfo info: vehicleInfoList) {
+                vehicleInfoMap.put(info.getId(), info);
+            }
+        } catch (FileNotFoundException e) {
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java]: Please do the same like you did with the ExcelFileDao. Just catch an **Exception** and throw IllegatlStateException. Do not prin anything to the STDOUT/STDERR

File: CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        } catch (JsonMappingException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    @Override
+    public List<VehicleInfo> getAllVehicles() {
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java]: Same comment as with the ExcelFileDao

File: CoteanuVlad/src/org/bootcamp/dao/VehicleInfoPlainFileDao.java [hunk 6]
```diff
@@ -6,55 +6,61 @@
 import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.io.InputStream;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
-import java.util.Scanner;
+import java.util.*;
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleInfoPlainFileDao.java]: Same comments as with the wild card imports

File: CoteanuVlad/src/org/bootcamp/dao/VehicleInfoPlainFileDao.java [hunk 7]
```diff
@@ -6,55 +6,61 @@
...
     private static final int VEHICLE_TYPE = 1;
     private static final int VEHICLE_AGE = 3;
     private static final int VEHICLE_MILES = 4;
     private static final int VEHICLE_IS_DIESEL = 5;
     private static final int VEHICLE_ID = 0;
     private static final int VEHICLE_FORMULA = 2;
 
-    private final String filePath;
+    private final VehicleInfo.Builder vehicleInfoBuilder = VehicleInfo.builder();
+    private Map<String, VehicleInfo> vehicleInfoMap;
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleInfoPlainFileDao.java]: Same comments for the Map field as with the other two Dao implementations

File: CoteanuVlad/src/org/bootcamp/dao/VehicleInfoPlainFileDao.java [hunk 8]
```diff
@@ -6,55 +6,61 @@
...
-    }
-
-    @Override
-    public List<VehicleInfo> getAllVehicles() {
-        final List<VehicleInfo> vehicles = new ArrayList<>();
-
-        final File inputFile = new File(this.filePath);
-
+        vehicleInfoMap = new HashMap<>();
+        VehicleInfo vehicleInfo = null;
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleInfoPlainFileDao.java]: You can declare this variable inside the **While**. As a best practive, avoid to explicitly initialize with **null**. It is redundant.

**Comment by AndreiOprisan94** [comment]
GG for being the first to submit Pull Request! 👌 👏

File: CoteanuVlad/src/org/bootcamp/dao/VehicleExcelFileDao.java [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+                            break;
+                        case VEHICLE_MILES:
+                            vehicleInfo.setNumberOfMiles((int)currentCell.getNumericCellValue());
+                            break;
+                        case VEHICLE_IS_DIESEL:
+                            vehicleInfo.setDiesel(currentCell.getBooleanCellValue());
+                            break;
+                    }
+
+                    if(currentCell.getAddress().getColumn() == VEHICLE_IS_DIESEL) {
```
[the remark this answers is not in the document]
Comment (mindit-coteanu-vlad) [reply CoteanuVlad/src/org/bootcamp/dao/VehicleExcelFileDao.java]: It's the last column of the excel row, case when I add the info to the map.

File: CoteanuVlad/src/org/bootcamp/dao/VehicleExcelFileDao.java [hunk 10]
```diff
@@ -6,12 +6,15 @@
...
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
 
-public class VehicleExcelFileDao implements VehicleInfoDao {
+public final class VehicleExcelFileDao implements VehicleInfoDao {
 
-    private final VehicleInfo.Builder vehicleInfoBuilder = VehicleInfo.builder();
-    private Map<String, VehicleInfo> vehicleInfoMap;
+    private Map<String, VehicleInfo> vehicleInfoMap = new HashMap<>();
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleExcelFileDao.java]: Make this one final to please.

File: CoteanuVlad/src/org/bootcamp/dao/VehicleExcelFileDao.java [hunk 11]
```diff
@@ -21,9 +24,7 @@
     private static final int VEHICLE_FORMULA = 2;
 
     public VehicleExcelFileDao(String filePath) {
-        vehicleInfoMap = new HashMap<>();
         try {
-
             InputStream inputStream = new FileInputStream(filePath);
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleExcelFileDao.java]: Make all this variables **final**. Where it applies.

File: CoteanuVlad/src/org/bootcamp/dao/VehicleExcelFileDao.java [hunk 12]
```diff
@@ -40,7 +41,7 @@ public VehicleExcelFileDao(String filePath) {
                     Cell currentCell = cellIterator.next();
                     switch (currentCell.getAddress().getColumn()) {
                         case VEHICLE_ID:
-                            vehicleInfo = new VehicleInfo();
+                            vehicleInfo = VehicleInfo.builder().build();
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleExcelFileDao.java]: Please use the **builder**. The right way.

File: CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java [hunk 13]
```diff
@@ -4,49 +4,41 @@
 import com.fasterxml.jackson.databind.JsonMappingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import org.bootcamp.model.VehicleInfo;
-import org.bootcamp.vehicle.Vehicle;
 
-import java.io.*;
+import java.io.File;
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java]: Remove the unused imports please.

File: CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java [hunk 14]
```diff
@@ -4,49 +4,41 @@
...
             }
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
-        } catch (JsonParseException e) {
-            e.printStackTrace();
-        } catch (JsonMappingException e) {
-            e.printStackTrace();
-        } catch (IOException e) {
-            e.printStackTrace();
+        } catch (Exception e) {
```
Comment (AndreiOprisan94) [inline CoteanuVlad/src/org/bootcamp/dao/VehicleJsonFileDao.java]: I think the right class name is **VehicleJsonFileDao** not **VehicleExcelFileDao**. Make the errorMessage **final** also

**Pull request merged**